### PR TITLE
Some tweaks to shed_util_common.

### PR DESCRIPTION
**Pull request by davebx**
I was going to include these in #1392, but then I decided it could be generally useful.

### CODE REVIEW

File: lib/tool_shed/util/shed_util_common.py [hunk 1]
```diff
@@ -431,16 +431,18 @@ def get_ordered_metadata_changeset_revisions( repository, repo, downloadable=Tru
...
         else:
-            rev = '-1'
-        changeset_tups.append( ( rev, changeset_revision ) )
-    sorted_changeset_tups = sorted( changeset_tups )
-    sorted_changeset_revisions = [ str( changeset_tup[ 1 ] ) for changeset_tup in sorted_changeset_tups ]
-    return sorted_changeset_revisions
+            rev = -1
+        changeset_tups.append( ( rev, repository_metadata.changeset_revision ) )
+    if sort_revisions:
+        changeset_tups = sorted( changeset_tups, key=itemgetter( 0 ), reverse=reverse )
```
Comment (nsoranzo) [inline lib/tool_shed/util/shed_util_common.py]: I think `changeset_tups.sort(key=itemgetter( 0 ), reverse=reverse)` would perform better.

Comment (davebx) [inline lib/tool_shed/util/shed_util_common.py]: I suspect you may be right.

**Comment by davebx** [comment]
This is WIP until I figure out the failing tests.

**Comment by martenson** [comment]
thanks @davebx !

**Pull request merged**